webapp/lib/controller.dart

```diff
@@ -157,25 +197,65 @@ void loadDashboardView() {
 }
 
 void loadPackageConfigurationView() {
-  var packages = model.packageConfigurationData.keys.toList();
+  _toggleNavActions(NavAction.ALLPROJECTS, false);
+  _toggleNavActions(NavAction.DASHBOARD, true);
+  var packages = new List<Map<String, String>>.from(model.activePackages.map((package) => new Map<String, String>.from({'id': package['id'], 'name': package['name']})));
   selectedPackage = router.routeParams['package'];
-  var configuratorView = new view.PackageConfiguratorView(packages, model.packageConfigurationData[selectedPackage]);
+  var configuratorView = new view.PackageConfiguratorView(packages, _findActivePackageConfigurationById(selectedPackage));
   view.contentView.renderView(configuratorView);
 }
 
 loadProjectConfigurationView() {
+  _toggleNavActions(NavAction.ALLPROJECTS, false);
+  _toggleNavActions(NavAction.DASHBOARD, true);
   view.navView.projectTitle = project?.projectName;
   view.navView.projectOrganizations = [''];
   view.contentView.renderView(new view.ProjectConfigurationView(model.projectConfigurationFormData, model.additionalProjectConfigurationLanguages));
 }
 
 void addPackage(String packageName) {
-  model.packageConfigurationData[packageName] = new model.Configuration()
-    ..availableTags = model.tags;
-  model.activePackages.add({'name': packageName, 'conversationsLink': '#/conversations', 'configurationLink': '#/package-configuration?package=$packageName',  'chartData': ''});
+  var id  = 'package-${new Uuid().v4().split('-')[0]}';
```

Comment:
consider turning this into a helper function since you need it a couple of times

Reply:
Created a helper method for it